interview-questions.md

```diff
@@ -7,90 +7,97 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: You would use super() when using class inheritance and you want to access the parent class and pull in all variables from the class.
 
-  Researched answer:
+  Researched answer: The super() method refers to a parent class. It is used to call the constructor of a parent class and to access those properties and methods that belong to that parent class. If you use "extends" on a class you must use super() to access the properties in methods in the parent class or else you will receive a reference error.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: The DOM stands for Document Object Model. It is how React can change the output on an application without reloading the entire page. The virtual DOM is what the changes are first logged to, it then gets updated virtually (in the background), and then the DOM is updated to display the change.
 
+  Researched answer: 
+  The DOM is a structured representation of HTML elements that is present in a web-app. It will contain a node for each HTML element as well. It allows developers to modify content via JavaScript. The Virtual DOM is a type of "lightweight" copy of the DOM. It is exactly the same as the DOM but it allows manipulation without the entire output on the screen being fully changed. Each time something is changed in a JSX file the virtual DOM is updated. It then compares the pre-updated DOM to the updated Virtual DOM, spots the difference and then makes those changes on the real DOM. It is a process known as "diffing".
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: A higher order function are functions that operate on other functions. They eitehr take other functions are arguments or return them.
 
-  Researched answer:
+  Researched answer: A higher order function is a function that accepts and/or returns another function. It is considered "higher-order" because instead of dealing with strings, numbers, or booleans, it takes it one step further to operate on functions. They can be used to store functions as variables, use them in arrays, assign them as methods, pass arguemnts, or return them from other functions. 
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is a combination of HTML and Javascript fully named JavaScript XML. It lets us write HTML in React.
```

Comment:
Absolutely, but not just in React! JSX is used in other technologies as well. It's really about allowing us to write HTML in Javascript.